provider/provider.go

```diff
@@ -412,16 +437,39 @@ func (p *dockerNativeProvider) Update(ctx context.Context, req *rpc.UpdateReques
 		return nil, errors.Wrapf(err, "diff failed because malformed resource inputs")
 	}
 
-	// if we are in Preview mode and buildOnPreview is set to false, return the news
-	if req.GetPreview() && !newInputs["buildOnPreview"].BoolValue() {
-		return &rpc.UpdateResponse{
-			Properties: req.GetNews(),
-		}, nil
-	}
-	// buildOnPreview needs all inputs to be resolved. Return error if trying to build on preview and there are Unknowns.
-	if req.GetPreview() && newInputs["buildOnPreview"].BoolValue() && newInputs.ContainsUnknowns() {
-		return nil, errors.New("cannot build on preview with unresolved inputs. " +
-			"Set buildOnPreview to False, or ensure all inputs are resolved at preview.")
+	if req.GetPreview() {
```

Comment:
Can we de-duplicate this block by parameterizing over its return type? It looks like this version will always return `rpc.UpdateResponce{Properties: req.GetNews()}, nil` or an error. The previous version will always return an `rpc.CreateResponce{Properties: req.GetProperties()}`, nil` or an error.